waspc/cli/Wasp/Cli/Command/Watch.hs

```diff
@@ -40,15 +43,41 @@ watch waspProjectDir outDir = FSN.withManager $ \mgr -> do
     listenForEvents :: Chan FSN.Event -> UTCTime -> IO ()
     listenForEvents chan lastCompileTime = do
       event <- readChan chan
-      let eventTime = FSN.eventTime event
-      if eventTime < lastCompileTime
-        then -- If event happened before last compilation started, skip it.
+      if isStaleEvent event lastCompileTime
+        then -- Ignore delayed/stale events older than our last compile time.
           listenForEvents chan lastCompileTime
         else do
+          -- Recompile, but only after a 1s period of no new events.
+          waitUntilNoNewEvents chan lastCompileTime 1
           currentTime <- getCurrentTime
           recompile
           listenForEvents chan currentTime
 
+    -- Blocks until no new events are recieved for a duration of `secondsToDelay`.
```

Comment:
Nice comment!
In Haskell, there is a tool called Haddock that parses comments and creates documentation from them. It introduces some special syntax to comments, you can find it in Haddock docs.
This is usually important for libraries and similar, it is not that important for our codebase, but still one useful thing is to use 
```
-- | This refers to the code below.
```

and

```
-- ^ This refers to code that is to the left (above?)
```

So it would be:
```hs
-- | Blocks until no new events are recieved for a duration of `secondsToDelay`.
-- Consumes any new events during an active timer window and then restarts wait.
-- If a stale event comes in during an active timer window, we immediately
-- return control to the caller.
```

Reply:
Cool, thanks for the tip on Haddock! I will update that 👍🏻

Reply:
Hmm, I tried updating it, but when I save the formatter removes the comment all-together when I use the Haddock style (both `-- |` and `-- ^`). I wonder if that is because it is a `where` function?

Reply:
Hmmmm possible, although I would expect it to work as a function header. Ok nm then!